src/warning.js

```diff
@@ -0,0 +1,101 @@
+QUnit.warning = ( function() {
+    var messages = [
+        {
+            type: "asyncTest",
+            message: "DEPRECATED: QUnit.asyncTest() will " +
+                "be removed in QUnit 2.0."
+        },
+        {
+            type: "init",
+            message: "DEPRECATED: QUnit.init() will throw " +
+                "an error in 2.0, and will be removed in 2.1."
+        },
+        {
+             type: "loggingCallbacks",
+             message: "DEPRECATED: Logging callbacks being assigned " +
+                "with new values instead of called as functions " +
+                "will be removed on QUnit 2.0.0."
+        },
+        {
+            type: "module-setup",
+            message: "DEPRECATED: beforeEach were previously named " +
+                "setup, which still exist and will be removed in QUnit 2.0.0."
+        },
+        {
+            type: "module-teardown",
+            message: "DEPRECATED: afterEach were previously named " +
+                "teardown, which still exist and will be removed in QUnit 2.0.0."
+        },
+        {
+            type: "push",
+            message: "DEPRECATED: QUnit.push() is deprecated and " +
+                "it's recommended to use it through its direct reference " +
+                "in the assertion context."
+        },
+        {
+            type: "reset",
+            message: "DEPRECATED: Use multiple tests instead of resetting " +
+                "inside a test. Use testStart or testDone for custom cleanup. " +
+                "This method will throw an error in 2.0, and will be removed in 2.1."
+        },
+        {
+            type: "start",
+            message: "DEPRECATED: The functionality of QUnit.start() will be " +
+                "altered. In QUnit 2.0, invoking it will ONLY affect the " +
+                "`QUnit.config.autostart` blocking behavior."
+        },
+        {
+            type: "stop",
+            message: "DEPRECATED: QUnit.stop() will be removed in QUnit 2.0."
+        },
+        {
+            type: "testDone-duration",
+            message: "DEPRECATED: duration property will be removed " +
+                "in 2.0.0, use runtime instead."
+        },
+        {
+            type: "exported-globals",
+            message: "DEPRECATED: QUnit 2.0 will not export its API " +
+                "methods to the window object."
+        }
+    ];
+
+    var once = [],
```

Comment:
To repost from the https://github.com/jquery/qunit/pull/918#discussion_r53466092 thread, this should be an object in which the keys correspond to message identifiers  and the values are plain objects with data to be determined.